Dockerfile

```diff
@@ -1,20 +1,23 @@
 # Define Mosquitto version
-ARG MOSQUITTO_VERSION=1.6.14
+ARG MOSQUITTO_VERSION=2.0.9
+# Define libwebsocket version
+ARG LWS_VERSION=2.4.2
 
 # Use debian:stable-slim as a builder for Mosquitto and dependencies.
 FROM debian:stable-slim as mosquitto_builder
 ARG MOSQUITTO_VERSION
+ARG LWS_VERSION
 
 # Get mosquitto build dependencies.
-RUN apt update && apt install -y wget build-essential cmake libssl-dev  libcjson-dev
+RUN set -ex; apt-get update; apt-get install -y wget build-essential cmake libssl-dev libcjson-dev
 
 # Get libwebsocket. Debian's libwebsockets is too old for Mosquitto version > 2.x so it gets built from source.
-RUN export LWS_VERSION=2.4.2  && \
-    wget https://github.com/warmcat/libwebsockets/archive/v${LWS_VERSION}.tar.gz -O /tmp/lws.tar.gz && \
-    mkdir -p /build/lws && \
-    tar --strip=1 -xf /tmp/lws.tar.gz -C /build/lws && \
-    rm /tmp/lws.tar.gz && \
-    cd /build/lws && \
+RUN set -ex; \
+    wget https://github.com/warmcat/libwebsockets/archive/v${LWS_VERSION}.tar.gz -O /tmp/lws.tar.gz; \
+    mkdir -p /build/lws; \
+    tar --strip=1 -xf /tmp/lws.tar.gz -C /build/lws; \
+    rm /tmp/lws.tar.gz; \
+    cd /build/lws; \
```

Comment:
Out of curiosity, why are you changing all instances of `&&` to `;` ?
I would think we would *WANT* the build to fail if any step had an error, but here it would keep going 🤔

Reply:
It's for readibility. `set -ex` will do the same thing by exiting in case of non zero exit code of any command in the script (and will print it out before executing)

Reply:
I'm primarily asking because I had not seen that construct before and was asking to learn myself. It's ultimately up to iegomez what he's comfortable with :)

Reply:
Oh dang, I have forgotten line [90](https://github.com/tetofonta/mosquitto-go-auth/blob/c1c0fdda86877a78e511ca6f13dc5ed8cb069544/Dockerfile#L90) =( You are right.

As for my understanding, docker executes all the run programs as `/bin/sh -c '<run command>'` (because if you leave a RUN entry without commands it errors out with `/bin/sh: 0: -c requires an argument`)

You can see this behavior of `set -e`/`set -ex` in a linux shell:
```
/bin/sh -c "set -ex; echo 'a'; [ 1 -eq 2 ]; echo 'b'"
``` 
this will run until ` [ 1 -eq 2 ]` returns non zero so it does not execute the second echo and it is the same as
```
/bin/sh -c "echo 'a' && [ 1 -eq 2 ] && echo 'b'"
```
I'm making a commit fixing those `&&`, sorry for the mistake :innocent:

Reply:
I'd actually leave it up to you guys, I'm not much of a Docker user and don't know what users normally expect. So if you both think the latest version of this PR is ready, I'll gladly merge it.

Reply:
So, @tetofonta, @kfdm, what do you say?

Reply:
Sorry for the time taken to answer.
I think this is an improvement on the current Dockerfile because of the double compilation of `libwebsocket` and because of the added functionality of the mosquitto utils.

As for the style, that's kind of a personal thing and does not matter. That's as you prefer, and I can revert those modifications without too much work.

Reply:
My comment was primarily that mixing build changes and code formatting all together makes the actual changes more difficult to see. Ideally it would have been two commits,

commit 1: add `set -ex` and change `&&` to `;` without changing any logic
commit 2: change the logic to change what is built or not built

My initial comment was mostly the surprise that there was a specific reason to change `&&` to `;`

Reply:
Aa said, I don't even use the Docker setup, so though I agree 2 commits with separate changes would be nice, I'm fine with squashing and merging if you don't have the time. I don't see us reverting the change anyway.

Reply:
@tetofonta can you do a rebase of your commit and split it up into logical commits that we can more easily review? If you want to change all the `&&` to `set -ex` and `;` then those need to be their own commit, so that the actual changes can be reviewed by themselves.

Reply:
Btw, @kfdm, @tetofonta , would you be interested in being maintainers of this project?
I ask because I don't look at it much these days, every few months I get some time and start looking at issues and PRs, but I could certainly rely on a couple more hands watching it.

No pressure though, feel free to say no. And I really mean that, helping with a particular issue is great help already, I'm not trying to drag you into something you might not want.

Cheers!

Reply:
I definitely want to see this project continue (and I'm using it in my [own project](https://github.com/kfdm/django-mqtt#mosquitto-go-auth) ) though Go is not a language I use on a regular basis, and the http auth is the only one I'm familiar with, so I'm not sure if I'm one of the most qualified. I could mostly only check minor Docker or http plugin requests.

That said, for the most part this library seems to do what it does well, and may not actually need a lot of attention besides small bug fixes, unless you had some other long term plans you wanted to attempt.

Reply:
He, seems like your project is private, I got a 404.
And yeah, that's fair, the project tends to be quiet most of the time until someone brings up some issue or a feature request and depending on the time I might not be too responsive. So let's see how it goes for now, and thank you very much for your response.

Reply:
Hmm, thought I had already made it public. It should be public now.